entities/src/main/java/com/flipkart/varadhi/entities/TopicResource.java

```diff
@@ -29,4 +45,13 @@ public TopicResource(
         this.capacityPolicy = capacityPolicy;
     }
 
+    public void validate() {
+        Set<ConstraintViolation<TopicResource>> violations = super.getValidator().validate(this);
+        if (violations.isEmpty()) {
+            return;
+        }
+        throw new ValidationException(violations.stream()
+                .map(violation -> violation.getPropertyPath() + ": " + violation.getMessage())
+                .collect(Collectors.joining(", ")));
+    }
```

Comment:
Does each resource/entity need to implement validate() ?
It seems validate() can remain outside and can be invoked on the respective objects. If yes, web\extensions.java might be better place to have these methods and do validation as part of asPojo() method.

Reply:
have moved it to a central place